src/pages/ReimbursementAccount/CompanyStep.js

```diff
@@ -46,138 +46,104 @@ class CompanyStep extends React.Component {
         }
 
-        if (!ValidationUtils.isValidUSPhone(this.state.companyPhone, true)) {
-            errors.companyPhone = true;
+        if (!values.incorporationDate || !ValidationUtils.isValidDate(values.incorporationDate)) {
+            errorTexts.incorporationDate = this.props.translate('bankAccount.error.incorporationDate');
         }
 
-        ReimbursementAccount.setBankAccountFormValidationErrors(errors);
+        if (!values.incorporationDate || !ValidationUtils.isValidPastDate(values.incorporationDate)) {
+            errorTexts.incorporationDateFuture = this.props.translate('bankAccount.error.incorporationDateFuture');
+        }
 
-        return _.size(errors) === 0;
-    }
+        if (!values.incorporationState) {
+            errorTexts.incorporationState = this.props.translate('bankAccount.error.incorporationState');
+        }
 
-    submit() {
-        if (!this.validate()) {
-            return;
+        if (!values.companyPhone || !ValidationUtils.isValidUSPhone(values.companyPhone, true)) {
+            errorTexts.companyPhone = this.props.translate('bankAccount.error.phoneNumber');
         }
 
-        const bankAccount = {
-            // Fields from BankAccount step
-            ...ReimbursementAccountUtils.getBankAccountFields(this.props, ['bankAccountID', 'routingNumber', 'accountNumber', 'bankName', 'plaidAccountID', 'plaidAccessToken', 'isSavings']),
+        if (!values.companyName) {
+            errorTexts.companyName = this.props.translate('bankAccount.error.companyName');
+        }
 
-            // Fields from Company step
-            ...this.state,
-            incorporationDate: moment(this.state.incorporationDate).format(CONST.DATE.MOMENT_FORMAT_STRING),
-            companyTaxID: this.state.companyTaxID.replace(CONST.REGEX.NON_NUMERIC, ''),
-            companyPhone: LoginUtils.getPhoneNumberWithoutUSCountryCodeAndSpecialChars(this.state.companyPhone),
-        };
+        if (!values.hasNoConnectionToCannabis) {
+            errorTexts.hasNoConnectionToCannabis = this.props.translate('bankAccount.error.restrictedBusiness');
+        }
 
-        BankAccounts.updateCompanyInformationForBankAccount(bankAccount);
+        const errors = {};
+        _.each(errorTexts, (key) => {
+            errors[key] = true;
+        });
+
+        BankAccounts.setBankAccountFormValidationErrors(errors);
+        BankAccounts.updateReimbursementAccountDraft(values);
+
+        return errorTexts;
+    }
+
+    submit(values) {
+        const incorporationDate = moment(values.incorporationDate).format(CONST.DATE.MOMENT_FORMAT_STRING);
+        BankAccounts.setupWithdrawalAccount({
+            ...values,
+            incorporationDate,
+            companyTaxID: values.companyTaxID.replace(CONST.REGEX.NON_NUMERIC, ''),
+            companyPhone: LoginUtils.getPhoneNumberWithoutUSCountryCodeAndSpecialChars(values.companyPhone),
+        });
```

Comment:
We should use the logic we have on [main](https://github.com/Expensify/App/blob/e2ac5e116f5d7522287ef3d126fee1749d4af664/src/pages/ReimbursementAccount/CompanyStep.js#L162-L173), since the API method has changed.

```suggestion
 const bankAccount = {
            // Fields from BankAccount step
            ...ReimbursementAccountUtils.getBankAccountFields(this.props, ['bankAccountID', 'routingNumber', 'accountNumber', 'bankName', 'plaidAccountID', 'plaidAccessToken', 'isSavings']),

            // Fields from Company step
            ...values,
            incorporationDate: moment(values.incorporationDate).format(CONST.DATE.MOMENT_FORMAT_STRING),
            companyTaxID: values.companyTaxID.replace(CONST.REGEX.NON_NUMERIC, ''),
            companyPhone: LoginUtils.getPhoneNumberWithoutUSCountryCodeAndSpecialChars(values.companyPhone),
        };

        BankAccounts.updateCompanyInformationForBankAccount(bankAccount);
```

Reply:
done